x-pack/platform/plugins/shared/ml/public/application/model_management/trained_models_service.ts

```diff
@@ -0,0 +1,600 @@
+    );
+  }
+
+  private updateUiStateForDeployment(modelId: string) {
+    const currentModels = this.modelItems;
+    const updatedModels = currentModels.map((model) =>
+      isBaseNLPModelItem(model) && model.model_id === modelId
+        ? { ...model, state: MODEL_STATE.STARTING }
+        : model
+    );
+    this._modelItems$.next(updatedModels);
+  }
+
+  private markDownloadAborted(modelId: string) {
+    this.abortedDownloads.add(modelId);
+  }
+
+  private mergeModelItems(
+    items: TrainedModelUIItem[],
+    spaces: Record<string, string[]>
+  ): TrainedModelUIItem[] {
+    const existingItems = this.modelItems;
+
+    return items.map((item) => {
+      const prevItem = existingItems.find((i) => i.model_id === item.model_id);
+      const baseItem = { ...item, spaces: spaces[item.model_id] };
+
+      if (!prevItem || !isBaseNLPModelItem(prevItem) || !isBaseNLPModelItem(item)) return baseItem;
+
+      if (prevItem.state === MODEL_STATE.DOWNLOADING) {
+        return { ...baseItem, state: prevItem.state, downloadState: prevItem.downloadState };
+      }
+
+      if (
+        prevItem.state === MODEL_STATE.STARTING &&
+        (this.deploymentInProgress.has(item.model_id) ||
+          this.activeDeployments.some((deployment) => deployment.modelId === item.model_id)) &&
+        item.state !== MODEL_STATE.STARTED
+      ) {
+        return { ...baseItem, state: prevItem.state };
+      }
+
+      return baseItem;
+    });
+  }
+
+  private setupFetchingSubscription() {
+    const fetchTrigger$ = merge(
+      this._reloadSubject$,
+      this._scheduledDeployments$.pipe(
+        pairwise(),
+        filter(([prev, curr]) => {
+          return prev.length !== curr.length;
+        })
+      )
+    );
+
+    this.subscription.add(
+      fetchTrigger$
+        .pipe(
+          tap(() => this._isLoading$.next(true)),
+          switchMap(() => {
+            const modelsList$ = from(this.trainedModelsApiService.getTrainedModelsList()).pipe(
+              catchError((error) => {
+                this.displayErrorToast?.(
+                  error,
+                  i18n.translate('xpack.ml.trainedModels.modelsList.fetchFailedErrorMessage', {
+                    defaultMessage: 'Error loading trained models',
+                  })
+                );
+                return of([] as TrainedModelUIItem[]);
+              })
+            );
+
+            const spaces$ = this.canManageSpacesAndSavedObjects
+              ? from(this.savedObjectsApiService.trainedModelsSpaces()).pipe(
+                  catchError(() => of({})),
+                  map(
+                    (spaces) =>
+                      ('trainedModels' in spaces ? spaces.trainedModels : {}) as Record<
+                        string,
+                        string[]
+                      >
+                  )
+                )
+              : of({} as Record<string, string[]>);
+
+            return combineLatest([modelsList$, spaces$]).pipe(
```

Comment:
`forkJoin` operator is an alternative to `Promise.allSettled` that used before for this
```suggestion
            return forkJoin([modelsList$, spaces$]).pipe(
```

Reply:
Done in https://github.com/elastic/kibana/pull/205699/commits/1b14cf9a9d0109b63d2a2c9d948599de7cb60ef4